aten/src/ATen/core/jit_type.h

```diff
@@ -560,7 +560,11 @@ struct DictType : public Type {
   }
 
   DEFINE_IS_SUBCLASS(DictType);
+
   bool isSubtypeOf(const TypePtr rhs) const override {
+    if (rhs->kind() != TypeKind::DictType) {
```

Comment:
I am unclear on why this would fix asan, can you clarify?

Reply:
It won't, #16534 has the correct fix